cli.js

```diff
@@ -1,84 +1,132 @@
-const commander = require('commander');
-// grab the version to keep things consistent
-const { version } = require('./package.json');
+const yargs = require('yargs');
+const moniker = require('moniker');
+
+const logger = require('./lib/logger');
+const { makeNameValid } = require('./lib/nameUtil');
 const { deployHandler, createHandler, invokeHandler,
-  logHandler, loginHandler, removeHandler, unknownHandler } = require('./lib');
+  logsHandler, loginHandler, removeHandler } = require('./lib');
 
-const actionWrapper = function actionWrapper(action, actionOptions) {
-  return async (options) => {
-    await action({
-      options: Object.assign({}, actionOptions, options, { path: commander.path }),
-      args: commander.args,
-      rawArgs: commander.rawArgs,
-      name: commander.args[commander.args.length - 1]._name,
-    });
-  };
-};
+const handleCommand = async function handleCommand(options, specificHandler) {
+  const numArgs = options._.length;
+  if (numArgs > 1) {
+    yargs.showHelp();
+    logger.error('Too many positional args given for command');
+    process.exit(1);
+  }
 
-// again not the cleanest solution but the practical one while
-// still using commander instead of yargs
-const namedActionHandler = function namedActionHandler(action) {
-  return async (funcName, options) => {
-    await action({
-      options: Object.assign({}, { function: funcName }, options, { path: commander.path }),
-      args: commander.args,
-      rawArgs: commander.rawArgs,
-      name: commander.args[commander.args.length - 1]._name,
-    });
-  };
+  if (!options.function) {
+    yargs.showHelp();
+    logger.error('A valid function name is a required argument');
+    if (options._[0] === 'create') {
+      logger.error('The "create" command allows for the --random(r) flag in lieu of a name');
+    }
+    process.exit(1);
+  }
+  await specificHandler(options);
+  process.exit(0);
 };
 
-commander
-  .version(version)
-  .description('Binaris command line interface')
-  .option('-p, --path <path>',
-    // eslint-disable-next-line quotes
-    'Use directory dir. "create" will create this directory if needed.');
-
-commander
-  .command('login')
-  .description('Login to your Binaris account using an API key')
-  .action(actionWrapper(loginHandler));
-
-commander
-  .command('logs <function>')
-  .description('Print the logs of a function')
-  .option('-t, --tail',
-    'Outputs logs in "tail -f" fashion')
-  .action(namedActionHandler(logHandler));
-
-commander
-  .command('create')
-  .description('Create a function from template')
-  .option('-f, --function <name>',
-    'Name of the function to generate. If omitted, a name will be chosen at random')
-  .action(actionWrapper(createHandler));
+yargs
+  .option('path', {
+    alias: 'p',
+    describe: 'Use directory dir. "create" will create this directory if needed.',
```

Comment:
This message confuses me. I don't understand what will happen with `dir`.

Reply:
Don't disagree but also this is just a 1:1 port from the old CLI. I'm not sure we want to keep this functionality anyway but if we do we should definitely change this message.